src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -45,11 +47,14 @@ private void AddExceptionDiagnostic(Diagnostic diagnostic)
         /// </summary>
         public async Task<ImmutableArray<Diagnostic>> GetAnalyzerDiagnosticsAsync()
         {
+            _driver.StartCompleteAnalysis(_analyzedSyntaxTrees, _cancellationToken);
+
             // Invoke GetDiagnostics to populate the compilation's CompilationEvent queue.
             // Discard the returned diagnostics.
             _compilation.GetDiagnostics(_cancellationToken);
```

Comment:
The compatibility council rejected a binary breaking change, which ruled out just adding a new parameter with a default value. Requiring a non-null value would make deprecating the old API problematic.